lib/bolt/plugin.rb

```diff
@@ -188,6 +198,13 @@ def add_module_plugin(plugin_name)
         config: config_for_plugin(plugin_name)
       }
 
+      # Make sure that the plugin's config is validated _before_ these checks.
+      # This way, we can fail early on invalid plugin config instead of _after_
+      # loading the modulepath (which can be expensive).
+      mod = modules[plugin_name]
+      raise PluginError::Unknown, plugin_name unless mod&.plugin?
```

Comment:
Since we've now done this check, `Bolt::Plugin::Module.load` can just take the single `module` instead of the `modules` list it currently takes.